### PR TITLE
remove some warning when compliling an external program with -Wextra

### PR DESCRIPTION
**Pull request by TeXitoi**
# Issue

When I compile a c++ project using `gcc -Wextra` and valhalla, I have some annoying warnings.

This PR fixes them.

## Tasklist

 - [x] Add tests (irrelevant)
 - [ ] Review - you must request approval to merge any PR to master
 - [x] Add #fixes with the issue number that this PR addresses (irrelevant)
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md) (irrelevant)
 - [x] Update relevant [documentation](docs/README.md) (irrelevant)

## Requirements / Relations

None.

### CODE REVIEW

File: valhalla/baldr/datetime.h [hunk 1]
```diff
@@ -169,9 +169,13 @@ bool is_restricted(const bool type,
  * @return Returns std::tm time structure. If the input string is not valid this method
  *         sets tm_year to 0.
  */
-static std::tm iso_to_tm(const std::string& iso) {
+static inline std::tm iso_to_tm(const std::string& iso) {
   // Create an invalid tm, then populate it from the ISO string using get_time
-  std::tm t = {0, -1, -1, -1, -1, 0, 0, 0};
```
Comment (kevinkreiser) [inline valhalla/baldr/datetime.h]: @TeXitoi what compiler are you using, does POD struct list initialization not work on your platform?

Comment (TeXitoi) [reply valhalla/baldr/datetime.h]: GCC 6.3, I have a warning saying that not every fields are initialised.

Comment (TeXitoi) [reply valhalla/baldr/datetime.h]: And glibc tm have a out of spec field.

Comment (TeXitoi) [reply valhalla/baldr/datetime.h]: Thus, adding 2 more zeros remove the warning, but is not portable.

Comment (TeXitoi) [reply valhalla/baldr/datetime.h]: Another solution that would fix my issue is to put this code in the `.cc`

Comment (kevinkreiser) [reply valhalla/baldr/datetime.h]: nah this is fine. we'll just have to note that this is a place where we cant use that due to some systems defining `tm` differently.

**Comment by TeXitoi** [comment]
Thanks!

**Pull request merged**